File: WnW_Base/src/main/java/de/jjl/wnw/base/rune/parser/WnWKeyboardInputParser.java
package de.jjl.wnw.base.rune.parser;

import de.jjl.wnw.base.input.WnWInput;
import de.jjl.wnw.base.input.WnWKeyboardInput;
import de.jjl.wnw.base.rune.WnWInputParser;
import de.jjl.wnw.base.rune.WnWRune;
import de.jjl.wnw.dev.rune.Runes;

public class WnWKeyboardInputParser implements WnWInputParser
{
	@Override
	public WnWRune parseInput(WnWInput input)
	{
		if(input instanceof WnWKeyboardInput)
		{
			return parsePath(((WnWKeyboardInput) input).getInput());
		}
		return null;
	}
	
	public WnWRune parsePath(String input)
	{
		return lookupRune(input);
	}
	
	public WnWRune lookupRune(String input)
	{
		return Runes.getRuneForString(input);
		
	}
}

File: WnW_Base/src/main/kotlin/de/jjl/wnw/base/rune/parser/WnWKeyboardInputParser.kt
package de.jjl.wnw.base.rune.parser

import de.jjl.wnw.base.rune.WnWInputParser
import de.jjl.wnw.base.input.WnWInput
import de.jjl.wnw.base.rune.WnWRune
import de.jjl.wnw.dev.conn.rune.*
import de.jjl.wnw.base.input.WnWKeyboardInput
import de.jjl.wnw.base.util.path.WnWPath

class WnWKeyboardInputParser : WnWInputParser
{
	override fun parseInput(input: WnWInput): WnWRune?
	{
		if(input is WnWKeyboardInput)
		{
			System.out.println(input.input);
			return parsePath(input.input)
		}
		return null
	}
	
	fun parsePath(input: String): WnWRune?
	{
		return lookupRune(input)
	}
	
	fun lookupRune(input: String): WnWRune?
	{
		return Runes.getRuneForString(input)
	}
}